### DISCUSSION TITLE
Beta Coefficient, Up/Down Beta, and Beta Convexity

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#268] [Indicators]
Community discussion about [Beta Coefficient](https://dotnet.stockindicators.dev/indicators/Beta/#content)

**Comment by GordonEldest** [comment]
Beta+ and Beta- are more interesting that Beta.
Beta itself is pretty much outdated as it is generally either based on previous year values, or  TTM meaning it integreate a full year.
no stocks is so regular theses days.
Can be used for Quant but not for Trade

**Reply by DaveSkender** [reply]
You have any reference I can read?  Not seeing much on Google Search.  Sounds interesting.

**Comment by GordonEldest** [comment]
Not really. The concept is very simple
When Negative correlation to the index used for Beta correlation then, the bucket is  accounted to Beta- when positive it's for Beta+
Beta+ + Beta- <> Beta

You can get the idea here for exemple
http://jonathankinlay.com/2020/01/beta-convexity/
or 
https://static.seekingalpha.com/uploads/2018/9/17/saupload_Beta.png

**Reply by DaveSkender** [reply]
I see.  I’ll add to the backlog. - #533 

Similar in concept to [Ulcer Index](https://daveskender.github.io/Stock.Indicators/indicators/UlcerIndex/#content), but for correlation instead of volatility.

**Reply by GordonEldest** [reply]
Yes and no, 
1- yes somewhere but not fully if you refere to binding it in convexity.
because Ulcer refere to a RiskFreeReturn that are factored in and is not a completely fixed value.
This is not the case in convexity. Of course stocks do move with change in Bond for exemple but the relationship is far from anything linear.

2- no if you keep them appart.
I recommend using independantly B+ and B- as part of a trending analysis. 
Binding them in another single number does reduce again the entropy caught during computation.
Exemple
if Market is going down, reducing exposure by closins position can be done using the order of the Beta- of the portfolio.
That kind of part of an intelligent trail stop 

My grain of salt

**Reply by DaveSkender** [reply]
I only meant similar in the concept of separating directional components ... I'll do more reading on B+/B- when it comes up in the backlog and might have more questions for you.

**Reply by DaveSkender** [reply]
To close the loop on this thread, [Beta Up/Down](https://dotnet.stockindicators.dev/indicators/Beta/#content) was added to this indicator in [v1.19.0](https://github.com/DaveSkender/Stock.Indicators/releases/tag/1.19.0%2B10), back in November 2021.

**Comment by cliffmcmanus** [comment]
possible new issue or may be designed this way.
Using BetaType.All Both the ReturnsEval and ReturnsMrktfields contain the same value
<List<BetaResult> betaResults1 = _quotes.GetBeta(_quotes, 200, BetaType.All).RemoveWarmupPeriods().ToList();>
Please advise  Thanks Cliff

**Reply by DaveSkender** [reply]
Hey Cliff, thanks for being a longtime supporter!

[Beta](https://dotnet.stockindicators.dev/indicators/Beta/#content) is unique in that it expects _two_ different quote datasets » one representing your evaluation quotes and the other typically representing "the market" (e.g. S&P 500) as a comparator.  When you supply the same quotes as shown in your syntax, it'll always show a beta of 1 and the same return values (a simple % change in price for each).

```csharp
// usage
IEnumerable<BetaResult> results = quotesEval
  .GetBeta(quotesMarket, lookbackPeriods, type);
```
